server/routes/api/subscriptions.ts

```diff
@@ -0,0 +1,241 @@
+import invariant from "invariant";
+import Router from "koa-router";
+import { ValidationError } from "@server/errors";
+import auth from "@server/middlewares/authentication";
+import { Subscription, Event, Document } from "@server/models";
+import { authorize } from "@server/policies";
+import { presentSubscription } from "@server/presenters";
+import { SubscriptionEvent } from "@server/types";
+import { assertPresent, assertUuid } from "@server/validation";
+import pagination from "./middlewares/pagination";
+
+/** Typical workflow involves
+ * 1. Declaring required states.
+ * 2. Deserializing request body.
+ * 3. CRUD fetch for required data model.
+ * 4. Checking authorization of actor on subject.
+ * 5. CRUD operation by actor on subject.
+ * 6. Ordering and pagination.
+ * 7. Setting response body.
+ */
+
+const router = new Router();
+
+const subscribableEventsDocument = ["documents.update"];
+
+router.post(
+  "subscriptions.list",
+
+  auth(),
+  pagination(),
+
+  async (ctx) => {
+    const { user } = ctx.state;
+    const { documentId, event } = ctx.body;
+
+    // Make sure `documentId` is accompanied
+    // with valid subsribable events.
+    if (documentId) {
+      if (!subscribableEventsDocument.includes(event)) {
+        throw ValidationError(
+          `Event ${event} is not subscribable for documentId ${documentId}`
+        );
+      }
+    }
+
+    const document = await Document.findByPk(documentId);
+
+    authorize(user, "listSubscription", document);
+
+    const subscriptions = await Subscription.findAll({
+      where: {
+        documentId: document.id,
+        event,
+      },
+      order: [["createdAt", "DESC"]],
+      offset: ctx.state.pagination.offset,
+      limit: ctx.state.pagination.limit,
+    });
+
+    ctx.body = {
+      pagination: ctx.state.pagination,
+      data: subscriptions.map(presentSubscription),
+    };
+  }
+);
+
+router.post(
+  "subscriptions.info",
+  auth({
+    optional: true,
+  }),
```

Comment:
```suggestion
  auth(),
```

Definitely not optional